### ISSUE TITLE
[Kotlin 2.0] Fix `Variant::toGodot` call after KT-68339 is fixed

### ISSUE THREAD

**Issue by chippmann** [utopia-rise/godot-kotlin-jvm#643]
In Kotlin 2.0 there is a bug with enum initialization order which gives false positives in our `Variant::toGodot` call. See: https://youtrack.jetbrains.com/issue/KT-68339

After https://youtrack.jetbrains.com/issue/KT-68339 is fixed, we need to undo the corresponding change made in https://github.com/utopia-rise/godot-kotlin-jvm/pull/641

**Comment by CedNaru** [comment]
I'll add that we can use the opportunity to rework the whole Variant.kt file.
This is way more complex than it should be. We created it when we weren't sure of how Godot was behaving.
For example, we check the null value for everything variant type, when it can only happen for the OBJECT type.
We also have things like those composite types (JVM_INT) that are not part of the original Godot variant and at the time were used by the API gen, which is no longer necessary now that it handles the casting itself (we will probably require changes in the entry gen)

**Comment by chippmann** [comment]
Although the linked issue is fixed, we still encounter this issue with kotlin `2.0.20` there are several open issues linked in the issue above which are not yet resolved.
We'll wait on those for now
